royalties-handler/src/reward_entries.rs

```diff
@@ -0,0 +1,90 @@
+elrond_wasm::imports!();
+
+use nft_minter::common_storage::PaymentsVec;
+
+pub const FIRST_ENTRY_ID: usize = 1;
+
+#[elrond_wasm::module]
+pub trait RewardEntriesModule:
+    crate::common_storage::CommonStorageModule + crate::token_balance::TokenBalanceModule
+{
+    #[only_owner]
+    #[endpoint(createNewRewardEntry)]
+    fn create_new_reward_entry(&self) {
+        let current_epoch = self.blockchain().get_block_epoch();
+        let last_claim_epoch = self.last_claim_epoch().get();
+        require!(
+            current_epoch == last_claim_epoch,
```

Comment:
What happens if the owner doesn't call the claim function each epoch? Maybe it's better to have a more permissive condition:
current_epoch >= last_claim_epoch

Reply:
Owner does not have to claim each epoch, they only have to claim at least in the same epoch they try to split rewards. So owner can for instance only claim in epochs 1, 2, 5, and then create rewards for epoch 5 only.